File: src/Advance1005.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

vector<string>dictionary;

void init_map()
{
    dictionary.clear();
    dictionary.push_back("zero");
    dictionary.push_back("one");
    dictionary.push_back("two");
    dictionary.push_back("three");
    dictionary.push_back("four");
    dictionary.push_back("five");
    dictionary.push_back("six");
    dictionary.push_back("seven");
    dictionary.push_back("eight");
    dictionary.push_back("nine");
}

int main()
{
    init_map();
    string num;
    cin>>num;
    int sum = 0;
    for(int i=0;i<num.size();i++)
        sum+=num[i]-'0';

    string answer;
    if(sum==0) answer=dictionary[0];
    else{
        answer=dictionary[sum%10];
        sum /= 10;
    }

    while(sum>0){
        answer = dictionary[sum%10]+" "+answer;
        sum /= 10;
    }
    cout<<answer<<endl;
    return 0;
}

File: src/pat/Advance1005.java
package pat;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

/**
 * Created by zq on 2016/8/16.
 */
public class Advance1005 {

    private static Map<Character,String> dictionary;

    private static void initMap(){
        dictionary = new HashMap<Character,String>();
        dictionary.put('0',"zero");
        dictionary.put('1',"one");
        dictionary.put('2',"two");
        dictionary.put('3',"three");
        dictionary.put('4',"four");
        dictionary.put('5',"five");
        dictionary.put('6',"six");
        dictionary.put('7',"seven");
        dictionary.put('8',"eight");
        dictionary.put('9',"nine");
    }

    public static void main(String[] arg){

        initMap();

        Scanner scanner = new Scanner(System.in);
        Long integer = scanner.nextLong();
        Long sum = Long.valueOf(0);
        while(integer>0){
            sum += integer%10;
            integer /= 10;
        }
        String string = String.valueOf(sum);
        for(int i=0;i<string.length();i++){
            System.out.print(dictionary.get(string.charAt(i)));
            if(i!=string.length()-1) System.out.print(" ");
            else System.out.print("\n");
        }
    }
}
